### PR TITLE
out-of-body-experience: tighter follow behavior for avatar RigidBody during high speed flying

### PR DESCRIPTION
**Pull request by AndrewMeadows**
* measure velocity of HMD in world-frame and adjust avatar's RigidBody's follow velocity accordingly

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: a10ae20bf7f516a505defbd49f4ca34d70882579 available at: https://gist.github.com/a65b5ad0fefb2b231a98884a8139aee0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8676-a10ae20.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8676-a10ae20.dmg

File: libraries/physics/src/CharacterController.cpp [hunk 1]
```diff
@@ -453,7 +460,27 @@ void CharacterController::setParentVelocity(const glm::vec3& velocity) {
...
+        float oldSpeed = _followVelocity.length();
+        const float VERY_SLOW_HOVER_SPEED = 0.25f;
+        if (oldSpeed / newSpeed > 100.0f && newSpeed < VERY_SLOW_HOVER_SPEED) {
+            // avatar is stopping quickly
+            // HACK: slam _followVelocity and _rigidBody velocity immediately
+            _followVelocity = newFollowVelocity;
+            _rigidBody->setLinearVelocity(_followVelocity);
+        } else {
+            const float blend = 0.2f;
+            _followVelocity = (1.0f - blend) * _followVelocity + blend * newFollowVelocity;
```
Comment (AndrewMeadows) [inline libraries/physics/src/CharacterController.cpp]: I think in this case a simple naive smoothing filter is ok.  I'm just using this to smooth out some noise in the velocity measurement.  A blend of 0.2 is actually quite fast so and should work for reasonable acceleration rates at expected frame rates.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 4e1c1aec0af7ce19a60120d7826ae04bd721f2f2 available at: https://gist.github.com/a65b5ad0fefb2b231a98884a8139aee0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8676-4e1c1ae.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8676-4e1c1ae.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: a443cd65f4b185a63bab6876afb9ba693cedbffa available at: https://gist.github.com/a65b5ad0fefb2b231a98884a8139aee0<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8676-a443cd6.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8676-a443cd6.dmg

**Pull request merged**